code/modules/research/techweb/_techweb.dm

```diff
@@ -206,6 +226,39 @@
 			return FALSE
 	return TRUE
 
+/datum/techweb/proc/have_experiments_for_node(datum/techweb_node/node)
+	. = TRUE
+	for (var/experiment_type in node.required_experiments)
+		if (!completed_experiments[experiment_type])
+			return FALSE
+
+/datum/techweb/proc/can_unlock_node(datum/techweb_node/node)
+	return can_afford(node.get_price(src)) && have_experiments_for_node(node)
+
+/datum/techweb/proc/add_experiment(experiment_type)
+	. = TRUE
+	// check active experiments for experiment of this type
+	for (var/i in available_experiments)
+		var/datum/experiment/E = i
+		if (E.type == experiment_type)
+			return FALSE
+	// check completed experiments for experiments of this type
+	for (var/i in completed_experiments)
+		var/datum/experiment/E = i
+		if (E == experiment_type)
+			return FALSE
+	available_experiments += new experiment_type()
+
+/datum/techweb/proc/add_experiments(list/experiment_list)
+	. = TRUE
+	for (var/e in experiment_list)
+		var/datum/experiment/E = e
+		. = . && add_experiment(E)
```

Comment:
```suggestion
	for (var/listed_experiment in experiment_list)
		var/datum/experiment/experiment = listed_experiment 
		. = . && add_experiment(experiment)
```
This line does not meet our contribution standards, specifically for [variable name compliance](https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names), and should be refactored.

Also, you might want to use the `as anything` key here, as listed in the [contribution guidelines](https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#avoid-unnecessary-type-checks-and-obscuring-nulls-in-lists)